nx/lib/nx.ex

```diff
@@ -16679,6 +16679,74 @@ defmodule Nx do
     |> as_type(opts[:type])
   end
 
+  @doc """
+  Creates an Nx-tensor from an already-allocated memory space.
+
+  This function should be used with caution, as it is very backend-specific.
+
+  The `backend` argument is the backend module (such as `Nx.BinaryBackend`).
+  `opaque_pointer` is the corresponding value that would be returned from
+  a call to `get_pointer/2`.
+
+  ## Options
+
+    * `:names` - refer to `tensor/2`
+    * `:backend_opts` - options specific to the backend for handling
+      the pointer.
+
+  ## Examples
+
+      Nx.from_pointer(<<1, 2, 3, 4>>, {:s, 64}, {1, 3})
+      #Nx.Tensor<
+        s64[1][3]
+        [
+          [10, 20, 30]
+        ]
+      >
+
+      Nx.from_pointer(<<5, 6, 7>>, {:s, 64}, {1, 3}, names: [nil, :col], backend_opts: [some: :option, another: :option])
+      #Nx.Tensor<
+        s64[1][col: 3]
+        [
+          [10, 20, 30]
+        ]
+      >
+  """
+  @doc type: :creation
+  def from_pointer(backend, opaque_pointer, type, shape, opts \\ []) when is_atom(backend) do
+    Nx.Shape.validate!(shape, :shape)
+    type = Nx.Type.normalize!(type)
+    opts = keyword!(opts, [:names, :backend_opts])
+    backend.from_pointer(opaque_pointer, type, shape, opts)
+  end
+
+  @doc """
+  Returns an opaque pointer for a given tensor.
+
+  Can be used in conjunction with `from_pointer/5` to share the same memory
+  for multiple tensors, as well as for interoperability with other programming
+  languages.
+
+  ## Options
+
+    All options are backend-specific
```

Comment:
```suggestion
  All options are backend-specific.
```

We may also nest everything under `pointer_options` to avoid confusion.